superset-frontend/src/explore/actions/hydrateExplore.ts

```diff
@@ -120,17 +121,21 @@ export const hydrateExplore =
       lastRendered: 0,
     };
 
+    const allDatasources = {
+      ...datasources,
+      [getDatasourceUid(initialDatasource)]: initialDatasource,
+    };
+
+    dispatch(initDatasources(allDatasources));
+
     return dispatch({
       type: HYDRATE_EXPLORE,
       data: {
         charts: {
           ...charts,
           [chartKey]: chart,
         },
-        datasources: {
-          ...datasources,
-          [getDatasourceUid(initialDatasource)]: initialDatasource,
-        },
+        datasources: allDatasources,
```

Comment:
Do we need this if we `initDatasources` in line 129?

Reply:
You're right, we don't need `datasources` in `HYDRATE_EXPLORE`. Removed. 
We need `initDatasources` because it's a different state branch handled by the datasources reducer.